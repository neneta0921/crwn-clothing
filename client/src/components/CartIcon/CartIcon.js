import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { toggleCartHidden } from '../../redux/cart/cartActions';
import { selectCartItemsCount } from '../../redux/cart/cartSelectors';

import { CartIconContainer, ItemCountContainer, ShoppingIcon } from './CartIconStyled';

export const CartIcon = ({ toggleCartHidden, itemCount }) => (
  <CartIconContainer className="cart-icon-container" onClick={toggleCartHidden}>
    <ShoppingIcon />
    <ItemCountContainer className="item-count-container">{itemCount}</ItemCountContainer>
  </CartIconContainer>
);

const mapDispatchToProps = (dispatch) => ({
  toggleCartHidden: () => dispatch(toggleCartHidden()),
});

const mapStateToProps = createStructuredSelector({
  itemCount: selectCartItemsCount,
});

export default connect(mapStateToProps, mapDispatchToProps)(CartIcon);
