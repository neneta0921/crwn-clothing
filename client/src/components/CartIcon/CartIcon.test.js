import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { CartIcon } from './CartIcon';

describe('CartIcon component', () => {
  let wrapper;
  let mockToggleCartHidden;

  beforeEach(() => {
    mockToggleCartHidden = jest.fn();
    const mockProps = {
      itemCount: 0,
      toggleCartHidden: mockToggleCartHidden,
    };
    wrapper = shallow(<CartIcon {...mockProps} />);
  });

  it('should render CartIcon component', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call toggleCartHidden when icon is clicked', () => {
    wrapper.find('CartIconContainer').simulate('click');
    expect(mockToggleCartHidden).toHaveBeenCalled();
  });

  it('should render the itemCount as the text', () => {
    const itemCount = parseInt(wrapper.find('ItemCountContainer').text());
    expect(itemCount).toBe(0);
  });
});
