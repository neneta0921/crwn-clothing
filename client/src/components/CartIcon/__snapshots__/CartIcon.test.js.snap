// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component should render CartIcon component 1`] = `
<div
  className="sc-bdfBwQ flWUgh cart-icon-container"
  onClick={[MockFunction]}
>
  <svg
    className="sc-gsTCUz ilxHWr"
  >
    shopping-bag.svg
  </svg>
  <span
    className="sc-dlfnbm gyrcfu item-count-container"
  >
    0
  </span>
</div>
`;
